internal/subshell/subshell.go

```diff
@@ -239,24 +244,21 @@ func DetectShell(cfg sscommon.Configurable) (string, string) {
 	return name, path
 }
 
-func detectShellWindows() string {
-	// Windows does not provide a way of identifying which shell we are running in, so we have to look at the parent
-	// process.
-
+func detectShellParent() string {
 	p, err := process.NewProcess(int32(os.Getppid()))
 	if err != nil && !errors.As(err, ptr.To(&os.PathError{})) {
-		panic(err)
+		logging.Error("Failed to get parent process: %v", err)
 	}
 
-	for p != nil {
+	for p != nil && p.Pid != 0 {
 		name, err := p.Name()
 		if err == nil {
-			if strings.Contains(name, "cmd.exe") || strings.Contains(name, "powershell.exe") {
+			if supportedShellName(name) {
```

Comment:
I believe this can be an absolute path, so not guaranteed to just be the filename.